core/templates/dev/head/pages/exploration-player-page/templates/information-card-modal.directive.html

```diff
@@ -1,7 +1,7 @@
 <md-card class="md-padding">
   <div ng-style="infoCardBackgroundCss" class="oppia-info-card-logo-thumbnail">
     <img ng-src="<[getStaticImageUrl(infoCardBackgroundImageUrl)]>" class="oppia-info-card-bg-image">
-    <h2 class="oppia-info-card-title"><[explorationTitle]></h2>
+    <h2 ng-style="titleWrapper()" class="oppia-info-card-title"><[explorationTitle]></h2>
```

Comment:
Hi @donosco98, have you tried a CSS solution for this? For example, we could use a word-break: break-all property and use span here with an appropriate heading class (see [bootstrap typography section](https://getbootstrap.com/docs/4.0/content/typography/)) instead of the h2 tag. Thanks!

Reply:
@kevintab95 `    <span ng-style="titleWrapper()" class="oppia-info-card-title h2"><[explorationTitle]></span>
` like this?

Reply:
Also seanlip once mentioned on this issues tab [here](https://github.com/oppia/oppia/issues/7430#issuecomment-524124985) that we should use a directive which finds how much width is required for the element which is what I tried doing in the function `titleWrapper()`?

Reply:
Yep, we could use the heading class like that. The idea is to get rid of the js calculation. So we could have something like this:
![image](https://user-images.githubusercontent.com/11008603/68883800-bdf88d00-0737-11ea-9dd3-5b1b19d3c625.png)

Notice I added the text-break class and used h4 instead of h2 (because h2 seemed a bit larger).

Reply:
Actually I guess you can leave the header - span thing as-is. I was just wondering if there was a way to use bootstrap classes or css ot achieve this instead of using JS for the calculation.

Reply:
> Also seanlip once mentioned on this issues tab [here](https://github.com/oppia/oppia/issues/7430#issuecomment-524124985) that we should use a directive which finds how much width is required for the element which is what I tried doing in the function `titleWrapper()`?

Oh Okay.. I understand the concern now. In that case, this looks fine. Thanks.

Reply:
Thanks @kevintab95, any other changes?

Reply:
Do you think adding "text-break" class in the parent div would be a drop-in solution? ie. Just adding that class in affected places should solve this. 
cc. @seanlip

Reply:
I'm fine with that as a general approach if we do the work beforehand to check that it is indeed a general approach that works in all cases. (If so, we should add that to any other relevant issues / somewhere on the wiki.)

Reply:
@kevintab95,  So should I add just the "text break class"?

Reply:
Considering that with the CSS approach we have had issues in the past where words were broken even thought they weren't overflowing, I think the directive approach should be fine. Thanks @donosco98.